velox/vector/arrow/Bridge.cpp

```diff
@@ -559,24 +568,39 @@ void exportMaps(
   out.children = holder.getChildrenArrays();
 }
 
+template <TypeKind kind>
+void flattenAndExportDictionary(
+    const BaseVector& vec,
+    const Selection& rows,
+    ArrowArray& out,
+    memory::MemoryPool* pool,
+    VeloxToArrowBridgeHolder& holder) {
+  using NativeType = typename velox::TypeTraits<kind>::NativeType;
+  auto flatVector = BaseVector::create<FlatVector<NativeType>>(
+      vec.type(), rows.count(), pool);
+  auto dictVector = vec.asUnchecked<DictionaryVector<NativeType>>();
+  rows.apply([&](vector_size_t row) {
+    if (dictVector->isNullAt(row)) {
+      flatVector->setNull(row, true);
+    } else {
+      flatVector->set(row, dictVector->valueAt(row));
+    }
+  });
+  holder.saveVector(flatVector);
+  exportBase(*flatVector, rows, out, pool);
```

Comment:
can we `exportFlat` directly here?

Reply:
No, we need the nulls exported as well. `exportFlat` only exports the values and not the nulls.

Reply:
Added an an optimized path for non-nulls.